### PR TITLE
Removed needless allocations and function calls, Rewrote Immuclient p…

### PR DESCRIPTION
**Pull request by volknanebo**
…ackage layout

### CODE REVIEW

File: pkg/client/cache/file_cache.go [hunk 1]
```diff
@@ -17,12 +17,13 @@ limitations under the License.
...
 import (
+	"io/ioutil"
+
 	"github.com/codenotary/immudb/pkg/api/schema"
 	"github.com/golang/protobuf/proto"
-	"io/ioutil"
 )
 
-const ROOT_FN = ".root-"
+const ROOT_FN = ".root"
```
Comment (mmeloni) [inline pkg/client/cache/file_cache.go:26]: Please  keep the server uuid separator

File: cmd/immuclient/commands/misc.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+	"os"
+	"strconv"
+	"time"
+
+	c "github.com/codenotary/immudb/cmd"
+	"github.com/codenotary/immudb/pkg/api/schema"
+	"github.com/spf13/cobra"
+)
+
+func (cl *commandline) consistency(cmd *cobra.Command) {
```
[the remark this answers is not in the document]
Comment (volknanebo) [reply cmd/immuclient/commands/misc.go]: +

> @volknanebo @mmeloni maybe it makes sense that the `consistency` and `inclusion` commands are groupped in a file named something like `tamperproofing.go`?

+

File: pkg/client/batch_request.go [hunk 3]
```diff
@@ -1,12 +1,9 @@
 /*
 Copyright 2019-2020 vChain, Inc.
```
Comment (mmeloni) [inline pkg/client/batch_request.go:2]: License can't be modified

File: pkg/client/client.go [hunk 4]
```diff
@@ -1,12 +1,9 @@
 /*
 Copyright 2019-2020 vChain, Inc.
-
```
Comment (mmeloni) [inline pkg/client/client.go:3]: License can't be modified

File: pkg/client/client_test.go [hunk 5]
```diff
@@ -1,12 +1,9 @@
 /*
 Copyright 2019-2020 vChain, Inc.
```
Comment (mmeloni) [inline pkg/client/client_test.go:2]: License can't be modified

File: pkg/client/options.go [hunk 6]
```diff
@@ -1,12 +1,9 @@
 /*
 Copyright 2019-2020 vChain, Inc.
-
```
Comment (mmeloni) [inline pkg/client/options.go:3]: License can't be modified

File: pkg/client/root_service.go [hunk 7]
```diff
@@ -1,12 +1,9 @@
 /*
 Copyright 2019-2020 vChain, Inc.
-
```
Comment (mmeloni) [inline pkg/client/root_service.go:3]: License can't be modified

File: pkg/client/timestamp_service.go [hunk 8]
```diff
@@ -1,12 +1,9 @@
 /*
 Copyright 2019-2020 vChain, Inc.
-
```
Comment (mmeloni) [inline pkg/client/timestamp_service.go:3]: License can't be modified

File: pkg/client/types.go [hunk 9]
```diff
@@ -1,12 +1,9 @@
 /*
 Copyright 2019-2020 vChain, Inc.
-
```
Comment (mmeloni) [inline pkg/client/types.go:3]: License can't be modified

File: pkg/gw/history_handler.go [hunk 10]
```diff
@@ -1,12 +1,9 @@
 /*
 Copyright 2019-2020 vChain, Inc.
-
```
Comment (mmeloni) [inline pkg/gw/history_handler.go:3]: License can't be modified

**Comment by padurean** [comment]
@volknanebo Please note that any command run with `immuclient` now fails: `immuClient` is `nil` in `commandline`.

**Comment by volknanebo** [comment]
> @volknanebo Please note that any command run with `immuclient` now fails: `immuClient` is `nil` in `commandline`.

fixed

**Comment by mmeloni** [comment]
merge done in #148

**Pull request closed without merging**